modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -331,6 +441,84 @@ public void onFailure(Exception e) {
+        assert forceMergeRequest.indices() != null && forceMergeRequest.indices().length == 1 : "DLM force merges one index at a time";
+        final String targetIndex = forceMergeRequest.indices()[0];
+        logger.info("DLM is issuing a request to force merge index [{}]", targetIndex);
+        client.admin().indices().forceMerge(forceMergeRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(ForceMergeResponse forceMergeResponse) {
+                if (RestStatus.OK.equals(forceMergeResponse.getStatus())) {
+                    if (forceMergeResponse.getFailedShards() > 0) {
+                        String message = Strings.format(
+                            "Force merge request resulted in %d failed shards",
+                            forceMergeResponse.getFailedShards()
+                        );
+                        listener.onFailure(new ElasticsearchException(message));
+                        logger.warn(message);
+                    } else if (forceMergeResponse.getTotalShards() != forceMergeResponse.getSuccessfulShards()) {
+                        String message = Strings.format(
+                            "Force merge request only had %d successful shards out of a total of %d",
+                            forceMergeResponse.getSuccessfulShards(),
+                            forceMergeResponse.getTotalShards()
+                        );
+                        listener.onFailure(new ElasticsearchException(message));
+                        logger.warn(message);
+                    } else {
+                        logger.info("DLM successfully force merged index [{}]", targetIndex);
+                        setForceMergeCompletedTimestamp(targetIndex, listener);
+                    }
+                } else {
+                    String message = Strings.format("Force merge request returned a %s response code", forceMergeResponse.getStatus());
+                    listener.onFailure(new ElasticsearchException(message));
+                    logger.warn(message);
+                }
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                String previousError = errorStore.getError(targetIndex);
+                listener.onFailure(e);
+                // To avoid spamming our logs, we only want to log the error once.
+                if (previousError == null || previousError.equals(errorStore.getError(targetIndex)) == false) {
+                    logger.warn(
+                        () -> Strings.format(
+                            "DLM encountered an error trying to force merge index [%s]. DLM will attempt to force merge the index on its "
+                                + "next run.",
+                            targetIndex
+                        ),
+                        e
+                    );
+                }
+            }
```

Comment:
Like above, maybe a nit, but should we do the logic that pertains to this method before we call `listener.onFailure(e)` ?